app/models/canonical_armors_smithing_material.rb

```diff
@@ -4,7 +4,6 @@ class CanonicalArmorsSmithingMaterial < ApplicationRecord
   belongs_to :canonical_armor
   belongs_to :canonical_material
 
-  validates :canonical_armor_id, presence: true
-  validates :canonical_material_id, presence: true
```

Comment:
Doing this really is redundant since the `belongs_to` association already takes care of it.